### PR TITLE
Add More Tests for `getPosts`

### PR DESCRIPTION
**Pull request by funnyboy-roks**

### CODE REVIEW

File: src/lib/testing/utils.ts [hunk 1]
```diff
@@ -37,13 +44,51 @@ export const testWithDb = <T>(
...
+            return new Date(ret);
+        },
+        take: (n: number) => {
+            return Array(n).fill(undefined).map(r.next);
+        },
+    };
+    return r;
+};
+
+export const assert_nonnull = <T>(t: T): t is NonNullable<T> => {
```
Comment (jackjohn7) [inline src/lib/testing/utils.ts:78]: nit: snake_case should be camelCase here.

File: src/lib/testing/utils.ts [hunk 2]
```diff
@@ -37,13 +44,51 @@ export const testWithDb = <T>(
...
+    if (t === null) {
+        throw new Error('Assertion failed.  t === null');
+    }
+    return true;
+};
+
+export const create_users = async (
```
Comment (jackjohn7) [inline src/lib/testing/utils.ts:85]: nit: snake_case should be camelCase here.

**Review by jackjohn7 — CHANGES_REQUESTED** [review]
Looks good so far. I'll be trying the user filtering on my branch. I like what you did with `testWithDb`

File: src/lib/testing/utils.ts (hunk 1 above)
Comment (funnyboy-roks) [reply src/lib/testing/utils.ts:78]: I can't help my inner Rust dev

Comment (funnyboy-roks) [reply src/lib/testing/utils.ts]: Actually removed this since it wasn't used anyways.

File: src/lib/testing/utils.ts (hunk 2 above)
Comment (funnyboy-roks) [reply src/lib/testing/utils.ts]: Fixed, but snake_case looks better :)

**Review by jackjohn7 — APPROVED** [review]
Looks great!

**Pull request merged**